framework/console/controllers/CacheController.php

```diff
@@ -270,6 +270,12 @@ private function findCaches(array $cachesNames = [])
                 $caches[$name] = $component['class'];
             } elseif (is_string($component) && $this->isCacheClass($component)) {
                 $caches[$name] = $component;
+            } elseif (is_callable($component)) {
```

Comment:
@cebe what do you think about removing the whole giant condition altogether and leaving a single call to the `Yii::$app->get($name)`. This *will* add some overhead in certain scenarios but the code will be significantly cleaner. After all this is just a maintenance console command.